src/dvc_data/hashfile/meta.py

```diff
@@ -64,8 +66,14 @@ def from_info(
 
     @classmethod
     def from_dict(cls, d: Dict[str, Any]) -> "Meta":
-        kwargs = {field: d[field] for field in fields_dict(cls) if field in d}
+        kwargs = {}
+        for field in cls.fields:
+            if field in d:
+                kwargs[field] = d[field]
```

Comment:
Turns out, for a small value of `cls.fields` as is the case here, for loops are faster than dict comprehension.